### PR TITLE
Testing Tic_Tac_Toe Ruby Code 

### PR DESCRIPTION
**Pull request by Davidosky007**
## Testing Tic_Tac_Toe Ruby Code with Rspec
In this project I
* I added specs for all public methods created in the tic tac toe game
* I wrapped spec for each method in describe blocks
* I used 'let' to define variables for test cases

### CODE REVIEW

**Review by tirthajyoti-ghosh — CHANGES_REQUESTED** [review]
# Changes required 🛑 

## What a remarkable project you have built 💯! This is impressive 🤩👌🧨!

- Your tests are very well-structured. Great job!
- Your README is very descriptive. I loved it.

### But in order to get it approved, we need to work on some changes. Let's see...

- It's very important that you test all the edge cases of your logic. 
  - For example, in your `validate_position` method in the `Game` class, you are checking if the `position` is between 1 and 9 **and** it is not a String. But in your test, you are only checking with numbers. 
  - Example 2, in the `game_status?` method in the `Game` class, you have three paths of execution - one for the `if` path, one for the `elsif` part, and one for the normal flow. But you are only testing the normal flow (when neither the `if` and the `elsif` is executed and `false` is returned).

  - [x] Please add tests for all of your logic.

- When you are writing tests to check all aspects of the method logic, you will find that you need to add tests for both the positive and negative scenarios. Right now, you are only testing positive scenarios. 

  A negative scenario means that when you pass the wrong input to your methods, your method should return a wrong output. Eg., let's say that you have a method that adds two numbers. A positive scenario is when you pass in two numbers and expect your method to return their sum. And the negative scenario would be when you pass in two numbers and **not** expect your method to return the difference of the two numbers.

  - [x] Please add both positive and negative scenarios for all your tests, especially, the logic methods (`win?`, `draw?`, etc.)

_Please note that the examples I have given are just some of the points I Have pointed out. Please go through your spec and implement the changes._

- The README currently doesn't have the testing instructions.
  - [x] Please add the testing instructions in your README.

- Please remove the `.vscode` folder from your repository and add it to `.gitignore`. There shouldn't be any irrelevant files committed.

Your project looks absolutely fantastic. Just make these changes and apply for another code review.

Happy coding 👨‍💻 

*Need help? Shoot me a message on Slack - [@Ghosh](https://app.slack.com/client/T47CT8XPG/DRDM6LAQN/user_profile/URB1MHXK6)*
<hr>

**Tirthajyoti Ghosh**

[GitHub](https://github.com/tirthajyoti-ghosh) | [LinkedIn](https://www.linkedin.com/in/tirthajyoti-ghosh/) | [Twitter](https://twitter.com/terrific_ghosh)

File: lib/game.rb [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      return true
+    elsif game_board.draw?
+      display_board(game_board.board)
+      puts ' The game is a tie'
+      puts ''
+      return true
+    end
+    false
+  end
+end
```
Comment (kobiyoyo) [inline lib/game.rb:91]: - [ ] All the public methods of this class should be tested ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/all_tic_tac_toe.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team     
#### Great Job so far, everything works fine, but there are a few more things required. 
### General
- [ ] Please remove the .vscode folder from your repository and add it to .gitignore. There shouldn't be any irrelevant files committed.
- [ ] It will be nice if the test file is divided into a separate file according to the classes
- [ ] Check comments below

### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Review by PhillipUg — APPROVED** [review]
# Status: Project Approved! 🥇 
Congratulations on implementing all project requirements and getting this project approved. :+1:

You can merge this branch into the development branch when you're ready and go on to your next project.
:bulb:  Do not forget to submit your project completion form after merging this branch.

All the best team,

For any assistance, you can contact me on slack as **Phillip Musiime**
